lldb/source/Plugins/Process/elf-core/ThreadElfCore.h

```diff
@@ -76,14 +76,33 @@ static_assert(sizeof(ELFLinuxPrStatus) == 112,
               "sizeof ELFLinuxPrStatus is not correct!");
 
 struct ELFLinuxSigInfo {
-  int32_t si_signo;
-  int32_t si_code;
+  int32_t si_signo; // Order matters for the first 3.
   int32_t si_errno;
+  int32_t si_code;
+  // Copied from siginfo_t so we don't have to include signal.h on non 'Nix
+  // builds, we add `g` to the si_ prefix because siginfo_t defines them as
+  // macros.
+  struct {
```

Comment:
We should still make an anonymous union here for if/when we add more entries that overlap like in `siginfo_t`. Since this isn't C code, we should also put the `_sigfault` at the top:
```
union {
  struct _sigfault {
```